src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,137 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    createClients.call(this);
```

Comment:
This seems like some unnecessary complexity here, especially since `createClients` is only used here and it's not like we're exporting it for the sake of testing or something.  Can we just instantiate the clients here?